app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Optional
+
+
 class Dictionary:
-    pass
+    initial_capacity: int = 8
+    load_factor: float = 2 / 3
+
+    class Node:
+        def __init__(self, key: Any, hash_value: int, value: Any) -> None:
+            self.key = key
+            self.hash_value = hash_value
+            self.value = value
+            self.next: Optional["Dictionary.Node"] = None
+
+    def __init__(self) -> None:
+        self.capacity = self.initial_capacity
+        self.size = 0
+        self.table: list[Optional[Dictionary.Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size >= self.capacity * self.load_factor:
+            self._resize()
+
+        hash_value: int = hash(key)
+        index: int = self._index(hash_value)
+
+        if not self.table[index]:
+            self.table[index] = self.Node(key, hash_value, value)
+        else:
+            node: Optional[Dictionary.Node] = self.table[index]
+            while node:
```

Comment:
you could make implementation without `next` Nodes which would simplify the whole thing.
So you could have one method to calculate index - it would go through the elements of a hash table and get the index of one that is not None and has matching hash and key in case if you want to retrieve. OR would set an element when it finds a suitable empty cell or update an element if key and hash is matching.. We could discuss more on QNA if you want or double check the implementation that Yuri was showing on the lecture